core/domain/change_domain.py

```diff
@@ -0,0 +1,170 @@
+                'Value for %s in cmd %s: %s is not allowed' % (
+                    attribute, cmd_name, actual_value))
+
+
+class BaseChange(object):
+    """Domain object for changes made to storage models' domain objects."""
+
+    OPTIONAL_CMD_ATTRIBUTE_NAMES = []
+    ALLOWED_COMMANDS = []
+    COMMON_ALLOWED_COMMANDS = [{
+        'name': base_models.VersionedModel.CMD_DELETE_COMMIT,
+        'required_attributes': [],
+        'optional_attributes': []
+    }]
+
+    def __init__(self, change_dict):
+        """Initializes a BaseChange object from a dict.
+
+        Args:
+            change_dict: dict. The dict containing cmd name and attributes.
+
+        Raises:
+            ValidationError: The given change_dict is not valid.
+        """
+        self.validate_dict(change_dict)
+
+        cmd_name = change_dict['cmd']
+        self.cmd = cmd_name
+
+        all_allowed_commands = (
+            self.ALLOWED_COMMANDS + self.COMMON_ALLOWED_COMMANDS)
+
+        cmd_attributes = []
+        for cmd in all_allowed_commands:
+            if cmd['name'] == cmd_name:
+                cmd_attributes = (
+                    cmd['required_attributes'] + cmd['optional_attributes'])
+                break
+
+        for attribute in cmd_attributes:
+            setattr(self, attribute, change_dict.get(attribute))
+
+    def validate_dict(self, change_dict):
+        """Checks that the command in change dict is valid for the domain
+        object.
+
+        Args:
+            change_dict: dict. A dict of changes with keys as a cmd and the
+            attributes of a command.
+
+        Raises:
+            ValidationError. The change dict does not contain the cmd key,
+                or the cmd name is not allowed for the Change domain object
+                or the command attributes are missing or extra.
+        """
+        if 'cmd' not in change_dict:
+            raise utils.ValidationError('Missing cmd key in change dict')
+
+        cmd_name = change_dict['cmd']
+
+        valid_cmd_specs = None
+
+        all_allowed_commands = (
+            self.ALLOWED_COMMANDS + self.COMMON_ALLOWED_COMMANDS)
+        for cmd in all_allowed_commands:
+            if cmd['name'] == cmd_name:
+                valid_cmd_specs = copy.deepcopy(cmd)
+                break
+
+        if not valid_cmd_specs:
+            raise utils.ValidationError('Command %s is not allowed' % cmd_name)
+
+        valid_cmd_specs.pop('name', None)
+
+        actual_cmd_specs = copy.deepcopy(change_dict)
+        actual_cmd_specs.pop('cmd', None)
+
+        validate_cmd(
+            cmd_name, valid_cmd_specs, actual_cmd_specs)
+
+    def to_dict(self):
+        """Returns a dict representing the BaseChange domain object.
+
+        Returns:
+            A dict, mapping all fields of BaseChange instance.
+        """
+        base_change_dict = {}
+        base_change_dict['cmd'] = self.cmd
+        for attribute_name in self.OPTIONAL_CMD_ATTRIBUTE_NAMES:
```

Comment:
I think this is not a correct way to do it -- we should get the type of command, find the allowed attrs for it, and use those only. Not iterate through a list of all possible attr names.

Reply:
Done.